src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -103,7 +103,13 @@ class InterceptedMethods extends MiniPhaseTransform {
       case Any_## =>
           poundPoundValue(qual)
       case Any_!= =>
-          qual.select(defn.Any_==).appliedToArgs(tree.args).select(defn.Boolean_!)
+        val lhs = qual.tpe
+        val rhs = tree.args.head.tpe
+        if (!(lhs <:< rhs || rhs <:< lhs)) {
+          ctx.warning(ex"comparing values of types ${lhs.widenDealias} and ${rhs.widenDealias} using `!=' will always yield true",
+              tree.pos)
+        }
+        qual.select(defn.Any_==).appliedToArgs(tree.args).select(defn.Boolean_!)
```

Comment:
I think the isNullable change is good. TermRefs should be nullable if their underlying typrs are nullable. So I propose to just keep this change and drop the rest.